handlers/authentication.go

```diff
@@ -9,30 +11,60 @@ import (
 	"github.com/kiali/kiali/log"
 )
 
+type AuthenticationHandler struct {
+	saToken string
+}
+
 type AuthInfo struct {
 	Strategy              string `json:"strategy"`
 	AuthorizationEndpoint string `json:"authorizationEndpoint,omitempty"`
 }
 
-func AuthenticationHandler(next http.Handler) http.Handler {
+func NewAuthenticationHandler() (AuthenticationHandler, error) {
+	// Read token from the filesystem
+	saToken, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/token")
+	if err != nil {
+		return AuthenticationHandler{}, err
+	} else {
+		return AuthenticationHandler{
+			saToken: string(saToken),
+		}, nil
+	}
+}
+
+func (ahandler *AuthenticationHandler) HandleUnauthenticated(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		context := context.WithValue(r.Context(), "token", "")
+		next.ServeHTTP(w, r.WithContext(context))
+	})
+}
+
+func (ahandler *AuthenticationHandler) Handle(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		statusCode := http.StatusOK
 		conf := config.Get()
 
+		//set the token to be the SA token. If we are using OpenShift OAuth then we will use the OAuth token instead
+		token := "" // ahandler.saToken;
```

Comment:
The comment says this code should be setting the token to the SA token, however that part is commented out and token is being set to an empty string "".